src/Features/CSharp/Portable/Completion/CompletionProviders/AttributeNamedParameterCompletionProvider.cs

```diff
@@ -34,55 +35,62 @@ internal override bool IsInsertionTrigger(SourceText text, int characterPosition
 
         public override async Task ProvideCompletionsAsync(CompletionContext context)
```

Comment:
Instead of doign this at this level.  why not do this at the caller of these methods?  i.e. have the services wrap the calls to teh providers.  Instead of needing each provider to wrap its own calls?

Reply:
That wouldn't help anything. The method body of every async method implicitly catches all exceptions (by compiler-generated try-catch-all) and thus causes the stack to be unwound.

Reply:
In order to get a good crash dump we need the full stack present when the filter executes.